profiler/profiler.go

```diff
@@ -151,25 +161,29 @@ func (p *profiler) enqueueUpload(bat batch) {
 
 // send takes profiles from the output queue and uploads them.
 func (p *profiler) send() {
-	defer close(p.exit)
 	for bat := range p.out {
+		if p.uploadFunc == nil {
+			continue
+		}
 		if err := p.uploadFunc(bat); err != nil {
 			log.Error("Failed to upload profile: %v\n", err)
+
+			if err == errOldAgent {
+				// Turn any future uploads into noops since we know they won't work.
+				p.uploadFunc = nil
+				// Trigger stop of the profiler (do it async otherwise we'd deadlock).
+				go p.stop()
```

Comment:
After you've nicely fixed the leaking goroutine in Start, this seems a bit yolo :)